add .gitignore

rmsc opened the pull request.

acxz commented:
There is no need to add `.gitignore` files to the AUR packages. You can imagine if every package had a .gitignore for src and pkg that would be quite painful.

rmsc commented:
I'm not sure what you mean by "painful", but it sure keeps `git status` clean. Anyway, it's really up to you.

rmsc commented:
Even if "there's no need" for it, here's the output of `git status` on my checkout. I hope you understand what I mean, and perhaps reconsider:

```bash
git status
On branch ogre-conflicts
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ogre-1.9/.SRCINFO
	modified:   ogre-1.9/PKGBUILD

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	ignition-fuel_tools/ignition-fuel_tools-7.0.0-1-x86_64.pkg.tar.zst
	ignition-fuel_tools/ignition-fuel_tools-7.0.0.tar.gz
	ignition-fuel_tools/pkg/
	ignition-fuel_tools/src/
	ignition-gazebo/ignition-gazebo-6.3.0-1-x86_64.pkg.tar.zst
	ignition-gazebo/ignition-gazebo-6.3.0.tar.gz
	ignition-gazebo/pkg/
	ignition-gazebo/src/
	ignition-gui/ignition-gui-6.2.0-1-x86_64.pkg.tar.zst
	ignition-gui/ignition-gui-6.2.0.tar.gz
	ignition-gui/pkg/
	ignition-gui/src/
	ignition-physics/ignition-physics-5.1.0-1-x86_64.pkg.tar.zst
	ignition-physics/ignition-physics-5.1.0.tar.gz
	ignition-physics/pkg/
	ignition-physics/src/
	ignition-rendering/0001-update-to-version-6.1.0-fix-building-with-ogre2.patch
	ignition-rendering/ignition-rendering-6.1.0-1-x86_64.pkg.tar.zst
	ignition-rendering/ignition-rendering-6.1.0.tar.gz
	ignition-rendering/pkg/
	ignition-rendering/src/
	ignition-sensors/ignition-sensors-6.0.1-1-x86_64.pkg.tar.zst
	ignition-sensors/ignition-sensors-6.0.1.tar.gz
	ignition-sensors/pkg/
	ignition-sensors/src/
	ignition-transport/ignition-transport11_11.0.0.tar.gz
	ignition-utils/ignition-utils-1.1.0-1-any.pkg.tar.zst
	ignition-utils/ignition-utils1_1.1.0.tar.gz
	ignition-utils/pkg/
	ignition-utils/src/
	sdformat/pkg/
	sdformat/sdformat-12.0.0-1-x86_64.pkg.tar.zst
	sdformat/sdformat-12.0.0.tar.gz
	sdformat/sdformat-12.3.0-1-x86_64.pkg.tar.zst
	sdformat/sdformat-12.3.0.tar.gz
	sdformat/src/

no changes added to commit (use "git add" and/or "git commit -a")
```
